DATABASECHANGELOGLOCK table creation fails in PostgreSQL when schema name requires quoting

### Search first

- [x] I searched and no similar issues were found

### Description

When using Liquibase with PostgreSQL and setting database.objectQuotingStrategy = ObjectQuotingStrategy.QUOTE_ALL_OBJECTS, the DATABASECHANGELOG table is correctly created in the quoted schema. However, the creation of the DATABASECHANGELOGLOCK table fails with a syntax error if the schema name begins with a digit and requires quoting.

Logs

```
25-10-20 21:31:25.710 DEBUG liquibase.logging.core.JavaLogger:37          : Failed to create or initialize the lock table, trying again, iteration 1 of 10

liquibase.exception.UnexpectedLiquibaseException: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: ERROR: syntax error at or near "b5d63f449304c05a291c4c27136ebb5"
  Position: 21
	at liquibase.lockservice.StandardLockService.isDatabaseChangeLogLockTableCreated(StandardLockService.java:246)
	at liquibase.lockservice.StandardLockService.init(StandardLockService.java:119)
	at liquibase.command.core.helpers.DatabaseChangelogCommandStep.checkLiquibaseTables(DatabaseChangelogCommandStep.java:165)
	at liquibase.command.core.helpers.DatabaseChangelogCommandStep.run(DatabaseChangelogCommandStep.java:88)
	at liquibase.command.CommandScope.lambda$execute$6(CommandScope.java:263)
	at liquibase.Scope.child(Scope.java:223)
	at liquibase.Scope.child(Scope.java:199)
	at liquibase.command.CommandScope.execute(CommandScope.java:251)
	at liquibase.Liquibase.lambda$update$0(Liquibase.java:216)
	at liquibase.Scope.lambda$child$0(Scope.java:214)
	at liquibase.Scope.child(Scope.java:223)
	at liquibase.Scope.child(Scope.java:213)
	at liquibase.Scope.child(Scope.java:192)
	at liquibase.Liquibase.runInScope(Liquibase.java:1366)
	at liquibase.Liquibase.update(Liquibase.java:205)
	at liquibase.Liquibase.update(Liquibase.java:188)
	at liquibase.Liquibase.update(Liquibase.java:175)
	at com.riseworld.pushservice.domain.tenant.TenantService.runLiquibaseChangeSet(TenantService.kt:94)
	at com.riseworld.pushservice.domain.tenant.TenantService.updateTenantSchemas(TenantService.kt:104)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:359)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:380)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:728)
	at com.riseworld.pushservice.domain.tenant.TenantService$$SpringCGLIB$$0.updateTenantSchemas(<generated>)
	at com.riseworld.pushservice.domain.tenant.TenantSchemaInitializer.run(TenantSchemaInitializer.kt:18)
	at org.springframework.boot.SpringApplication.lambda$callRunner$4(SpringApplication.java:784)
	at org.springframework.util.function.ThrowingConsumer$1.acceptWithException(ThrowingConsumer.java:82)
	at org.springframework.util.function.ThrowingConsumer.accept(ThrowingConsumer.java:60)
	at org.springframework.util.function.ThrowingConsumer$1.accept(ThrowingConsumer.java:86)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:796)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:784)
	at org.springframework.boot.SpringApplication.lambda$callRunners$3(SpringApplication.java:772)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:510)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:772)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:325)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1361)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1350)
	at com.riseworld.pushservice.PushServiceApplicationKt.main(PushServiceApplication.kt:15)
Caused by: liquibase.exception.DatabaseException: org.postgresql.util.PSQLException: ERROR: syntax error at or near "b5d63f449304c05a291c4c27136ebb5"
  Position: 21
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:172)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:156)
	at liquibase.database.core.DatabaseUtils.initializeDatabase(DatabaseUtils.java:63)
	at liquibase.database.core.PostgresDatabase.rollback(PostgresDatabase.java:419)
	at liquibase.snapshot.SnapshotGeneratorFactory.checkLiquibaseTablesExistence(SnapshotGeneratorFactory.java:175)
	at liquibase.snapshot.SnapshotGeneratorFactory.checkExistence(SnapshotGeneratorFactory.java:252)
	at liquibase.snapshot.SnapshotGeneratorFactory.has(SnapshotGeneratorFactory.java:220)
	at liquibase.snapshot.SnapshotGeneratorFactory.hasDatabaseChangeLogLockTable(SnapshotGeneratorFactory.java:425)
	at liquibase.lockservice.StandardLockService.isDatabaseChangeLogLockTableCreated(StandardLockService.java:244)
	... 49 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: syntax error at or near "b5d63f449304c05a291c4c27136ebb5"
  Position: 21
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2733)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2420)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:372)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:517)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:434)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:194)
	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:180)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:168)
	... 57 common frames omitted

25-10-20 21:31:25.712  INFO liquibase.logging.core.JavaLogger:37          : Logging exception.
ERROR: Exception Details
ERROR: Exception Primary Class:  PSQLException
ERROR: Exception Primary Reason:  ERROR: syntax error at or near "b5d63f449304c05a291c4c27136ebb5"
  Position: 21
ERROR: Exception Primary Source:  PostgreSQL 13.21
```

### Steps To Reproduce

Create the database schema CREATE SCHEMA IF NOT EXISTS "4b5d63f449304c05a291c4c27136ebb5" and then try to run some change log with newly created schema.
```
          val database = DatabaseFactory.getInstance().findCorrectDatabaseImplementation(JdbcConnection(connection))
            database.defaultSchemaName = "4b5d63f449304c05a291c4c27136ebb5"
            database.objectQuotingStrategy = ObjectQuotingStrategy.QUOTE_ALL_OBJECTS
            val liquibase = Liquibase(
                "db/changelog/db.changelog-master.yaml",
                ClassLoaderResourceAccessor(resourceLoader.classLoader),
                database,
            )
            liquibase.update()
```

### Expected/Desired Behavior

Liquibase should consistently apply ObjectQuotingStrategy.QUOTE_ALL_OBJECTS to all internal table operations, including DATABASECHANGELOGLOCK, especially when schema names require quoting due to PostgreSQL syntax rules (e.g. names starting with digits).

### Liquibase Version

5.0.1, 4.30.0

### Database Vendor & Version

PostgreSQL 13.21

### Liquibase Integration

_No response_

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

Ubuntu 25.04

### Additional Context

Problematic code in `DatabaseUtils.java`
```
            if (database instanceof PostgresDatabase && defaultSchemaName != null) {
                String searchPath = executor.queryForObject(new RawParameterizedSqlStatement("SHOW SEARCH_PATH"), String.class);

                if (!searchPath.equals(defaultCatalogName) && !searchPath.equals(defaultSchemaName) && !searchPath.equals("\"" + defaultSchemaName + "\"") && !searchPath.startsWith(defaultSchemaName + ",") && !searchPath.startsWith("\"" + defaultSchemaName + "\",")) {
                    //
                    // Escape the default schema name if preserve schema case is set or if the schema name contains an @ symbol
                    //
                    String finalSearchPath;
                    if (Boolean.TRUE.equals(GlobalConfiguration.PRESERVE_SCHEMA_CASE.getCurrentValue()) || defaultSchemaName.contains("@")) {
                        finalSearchPath = ((PostgresDatabase) database).quoteObject(defaultSchemaName, Schema.class);
                    } else {
                        finalSearchPath = defaultSchemaName;
                    }
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

--- a/liquibase-integration-tests/src/test/java/liquibase/dbtest/AbstractIntegrationTest.java
+++ b/liquibase-integration-tests/src/test/java/liquibase/dbtest/AbstractIntegrationTest.java
@@ -17,6 +17,7 @@ import liquibase.command.core.helpers.DbUrlConnectionArgumentsCommandStep;
 import liquibase.database.Database;
 import liquibase.database.DatabaseConnection;
 import liquibase.database.DatabaseFactory;
+import liquibase.database.ObjectQuotingStrategy;
 import liquibase.database.core.*;
 import liquibase.database.jvm.JdbcConnection;
 import liquibase.datatype.DataTypeFactory;
@@ -103,6 +104,7 @@ public abstract class AbstractIntegrationTest {
     private final String indexWithAssociatedWithChangeLog;
     private Database database;
     private String defaultSchemaName;
+    private ObjectQuotingStrategy objectQuotingStrategy;
 
     private final String pathChangeLog;
 
@@ -158,6 +160,11 @@ public abstract class AbstractIntegrationTest {
         // If we should test with a custom defaultSchemaName:
         if (getDefaultSchemaName() != null && getDefaultSchemaName().length() > 0) {
             database.setDefaultSchemaName(getDefaultSchemaName());
+        }
+
+        // If we should test with a custom QUOTING STRATEGY
+        if(getObjectQuotingStrategy() != null) {
+            database.setObjectQuotingStrategy(getObjectQuotingStrategy());
         }
 
         SnapshotGeneratorFactory.resetAll();
@@ -1201,6 +1208,14 @@ public abstract class AbstractIntegrationTest {
 
     public void setDefaultSchemaName(String defaultSchemaName) {
         this.defaultSchemaName = defaultSchemaName;
+    }
+
+    protected ObjectQuotingStrategy getObjectQuotingStrategy() {
+        return objectQuotingStrategy;
+    }
+
+    protected void setObjectQuotingStrategy(ObjectQuotingStrategy objectQuotingStrategy) {
+       this.objectQuotingStrategy = objectQuotingStrategy;
     }
 
     @Test

--- a/liquibase-integration-tests/src/test/java/liquibase/dbtest/pgsql/PostgreSQLIntegrationTest.java
+++ b/liquibase-integration-tests/src/test/java/liquibase/dbtest/pgsql/PostgreSQLIntegrationTest.java
@@ -17,6 +17,7 @@ import liquibase.command.core.helpers.DatabaseChangelogCommandStep;
 import liquibase.command.core.helpers.DbUrlConnectionArgumentsCommandStep;
 import liquibase.database.Database;
 import liquibase.database.DatabaseFactory;
+import liquibase.database.ObjectQuotingStrategy;
 import liquibase.database.jvm.JdbcConnection;
 import liquibase.dbtest.AbstractIntegrationTest;
 import liquibase.diff.DiffGeneratorFactory;
@@ -45,8 +46,7 @@ import java.sql.*;
 import java.util.*;
 import java.util.concurrent.CompletableFuture;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
 import static org.junit.Assume.assumeNotNull;
 import static org.junit.Assume.assumeTrue;
 
@@ -55,6 +55,8 @@ public class PostgreSQLIntegrationTest extends AbstractIntegrationTest {
     private String dependenciesChangeLog;
     private String blobChangeLog;
     private String sleepChangelog;
+    private String rollbackEscapeChangelog;
+    private String createNumericSchema;
     private static DatabaseTestSystem localTestSystem;
 
     public PostgreSQLIntegrationTest() throws Exception {
@@ -62,6 +64,8 @@ public class PostgreSQLIntegrationTest extends AbstractIntegrationTest {
         dependenciesChangeLog = "changelogs/pgsql/complete/testFkPkDependencies.xml";
         blobChangeLog = "changelogs/pgsql/complete/testBlob.changelog.xml";
         sleepChangelog = "changelogs/pgsql/pg_sleep.sql";
+        rollbackEscapeChangelog = "changelogs/pgsql/rollback/rollback-escape.xml";
+        createNumericSchema = "changelogs/pgsql/rollback/rollback-create-numeric-schema.xml";
         localTestSystem = testSystem;
     }
 
@@ -434,6 +438,15 @@ public class PostgreSQLIntegrationTest extends AbstractIntegrationTest {
         }
     }
 
+    private void assertTableNotExists(String tableName) throws SQLException {
+        Connection connection = testSystem.getConnection();
+        DatabaseMetaData dbm = connection.getMetaData();
+        String catalog = connection.getCatalog();
+        try (ResultSet tables = dbm.getTables(catalog.toLowerCase(), null, tableName.toLowerCase(), null)) {
+            assertFalse("Table " + tableName + " should not be found, but is found.", tables.isBeforeFirst());
+        }
+    }
+
     private void dropTablesOutsidefastCheck() throws SQLException {
         Connection connection = testSystem.getConnection();
         try (Statement statement = connection.createStatement()) {
@@ -475,5 +488,26 @@ public class PostgreSQLIntegrationTest extends AbstractIntegrationTest {
         Assert.assertTrue("Using index function should be \"" + using + "\"", found);
     }
 
+    @Test
+    public void rollbackRestoresSearchPathForNumericSchema() throws Exception {
+        final String numericSchema = "4b5d63f449304c05a291c4c27136ebb5";
+        assumeNotNull(this.getDatabase());
+        this.setObjectQuotingStrategy(ObjectQuotingStrategy.QUOTE_ALL_OBJECTS);
+        Liquibase liquibase = createLiquibase(createNumericSchema);
+        liquibase.update("");
+        this.setDefaultSchemaName(numericSchema);
+
+        liquibase = createLiquibase(rollbackEscapeChangelog);
+
+        liquibase.update("");
+        assertTableExists("small_table");
+
+        liquibase.rollback(1, "");
+
+        assertTableExists("databasechangelog");
+        assertTableExists("databasechangeloglock");
+        assertTableNotExists("small_table");
+        clearDatabase();
+    }
 
 }

--- a/liquibase-integration-tests/src/test/resources/changelogs/pgsql/rollback/rollback-create-numeric-schema.xml
+++ b/liquibase-integration-tests/src/test/resources/changelogs/pgsql/rollback/rollback-create-numeric-schema.xml
@@ -1,0 +1,12 @@
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="
+          http://www.liquibase.org/xml/ns/dbchangelog
+          http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.27.xsd">
+
+    <changeSet id="1" author="rohan">
+        <sql>CREATE SCHEMA "4b5d63f449304c05a291c4c27136ebb5"</sql>
+    </changeSet>
+
+</databaseChangeLog>

--- a/liquibase-integration-tests/src/test/resources/changelogs/pgsql/rollback/rollback-escape.xml
+++ b/liquibase-integration-tests/src/test/resources/changelogs/pgsql/rollback/rollback-escape.xml
@@ -1,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
+    <preConditions>
+        <dbms type="postgresql"/>
+    </preConditions>
+
+    <changeSet id="22" author="rohan">
+        <createTable tableName="small_table">
+            <column name="id" type="int"/>
+        </createTable>
+    </changeSet>
+
+</databaseChangeLog>

--- a/liquibase-standard/src/main/java/liquibase/database/core/PostgresDatabase.java
+++ b/liquibase-standard/src/main/java/liquibase/database/core/PostgresDatabase.java
@@ -416,7 +416,7 @@ public class PostgresDatabase extends AbstractJdbcDatabase {
         // SET SEARCH_PATH SQL statements
         final Executor executor = Scope.getCurrentScope().getSingleton(ExecutorService.class).getExecutor("jdbc", this);
         if(executor.updatesDatabase()) {
-            DatabaseUtils.initializeDatabase(getDefaultCatalogName(), getDefaultSchemaName(), this);
+            DatabaseUtils.initializeDatabase(this.escapeObjectName(getDefaultCatalogName(), Catalog.class), this.escapeObjectName(getDefaultSchemaName(), Schema.class), this);
         }
     }
 

--- a/liquibase-standard/src/test/java/liquibase/database/core/PostgresDatabaseTest.java
+++ b/liquibase-standard/src/test/java/liquibase/database/core/PostgresDatabaseTest.java
@@ -7,6 +7,7 @@ import liquibase.database.Database;
 import liquibase.database.ObjectQuotingStrategy;
 import liquibase.exception.DatabaseException;
 import liquibase.structure.DatabaseObject;
+import liquibase.structure.core.Schema;
 import liquibase.structure.core.Table;
 import liquibase.structure.core.View;
 import liquibase.util.StringUtil;
@@ -158,6 +159,23 @@ public class PostgresDatabaseTest extends AbstractJdbcDatabaseTest {
         final String expectedPrimaryKeyName = "name_" + StringUtil.repeat("\u03A9", 15) + "_pkey";
 
         assertPrimaryKeyName(expectedPrimaryKeyName, this.database.generatePrimaryKeyName(nameWith15NonAsciiSymbols));
+    }
+
+    @Test
+    void rollbackUsesEscapedSchemaName() {
+        PostgresDatabase database = new PostgresDatabase();
+        database.setObjectQuotingStrategy(ObjectQuotingStrategy.QUOTE_ALL_OBJECTS);
+        database.setDefaultSchemaName("4b5d63f449304c05a291c4c27136ebb5");
+
+        String escaped = database.escapeObjectName(
+                database.getDefaultSchemaName(),
+                Schema.class
+        );
+
+        assertEquals(
+                "\"4b5d63f449304c05a291c4c27136ebb5\"",
+                escaped
+        );
     }
 
 //    @Test

